src/ZeroLog/JsonWriter.cs

```diff
@@ -0,0 +1,319 @@
+                    stringBuffer.Append(*(long*)dataPointer, StringView.Empty);
+                    dataPointer += sizeof(long);
+                    break;
+
+                case ArgumentType.Single:
+                    stringBuffer.Append(*(float*)dataPointer, StringView.Empty);
+                    dataPointer += sizeof(float);
+                    break;
+
+                case ArgumentType.Double:
+                    stringBuffer.Append(*(double*)dataPointer, StringView.Empty);
+                    dataPointer += sizeof(double);
+                    break;
+
+                case ArgumentType.Decimal:
+                    stringBuffer.Append(*(decimal*)dataPointer, StringView.Empty);
+                    dataPointer += sizeof(decimal);
+                    break;
+
+                case ArgumentType.Guid:
+                    stringBuffer.Append('"');
+                    stringBuffer.Append(*(Guid*)dataPointer, StringView.Empty);
+                    stringBuffer.Append('"');
+                    dataPointer += sizeof(Guid);
+                    break;
+
+                case ArgumentType.DateTime:
+                    stringBuffer.Append('"');
+                    stringBuffer.Append(*(DateTime*)dataPointer, StringView.Empty);
+                    stringBuffer.Append('"');
+                    dataPointer += sizeof(DateTime);
+                    break;
+
+                case ArgumentType.TimeSpan:
+                    stringBuffer.Append('"');
+                    stringBuffer.Append(*(TimeSpan*)dataPointer, StringView.Empty);
+                    stringBuffer.Append('"');
+                    dataPointer += sizeof(TimeSpan);
+                    break;
+
+                case ArgumentType.Enum:
+                    var enumArg = (EnumArg*)dataPointer;
+                    dataPointer += sizeof(EnumArg);
+                    enumArg->AppendTo(stringBuffer);
+                    break;
+
+                case ArgumentType.Null:
+                    stringBuffer.Append("null");
+                    break;
+
+                default:
+                    throw new ArgumentOutOfRangeException();
+            }
+        }
+
+        private static bool NeedsEscaping(string s)
+        {
+            // Not using linq here since it (might) add overhead.
+            foreach (var c in s)
+            {
+                if (c < '\u001F' || c == '"' || c == '/')
+                {
+                    return true;
+                }
+            }
+
+            return false;
+        }
+
+        private static void AppendEscapedJson(char c, StringBuffer stringBuffer)
+        {
+            // Escape characters based on https://tools.ietf.org/html/rfc7159
+            switch (c)
+            {
+                case '/':
+                    stringBuffer.Append("\\/");
+                    break;
+
+                case '"':
+                    stringBuffer.Append("\\\"");
+                    break;
+
+                case '\u0000':
```

Comment:
I think I'd prefer an `if (c < 32)` condition, with special casing for `\t`, `\n` etc and some code to handle the rest (which should almost never happen). This `switch` is kind of long...